nuxeo-features/nuxeo-annotation/nuxeo-annotation-rest-api/src/main/java/org/nuxeo/ecm/restapi/server/jaxrs/annotation/AnnotationAdapter.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * (C) Copyright 2018 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Funsho David
+ *
+ */
+
+package org.nuxeo.ecm.restapi.server.jaxrs.annotation;
+
+import java.util.List;
+
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.nuxeo.ecm.annotation.Annotation;
+import org.nuxeo.ecm.annotation.AnnotationService;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.webengine.model.WebAdapter;
+import org.nuxeo.ecm.webengine.model.impl.DefaultAdapter;
+import org.nuxeo.runtime.api.Framework;
+
+/**
+ * @since 10.1
+ */
+@WebAdapter(name = AnnotationAdapter.NAME, type = "annotationAdapter")
+@Produces(MediaType.APPLICATION_JSON)
+public class AnnotationAdapter extends DefaultAdapter {
+
+    public static final String NAME = "annotation";
+
+    @POST
+    public Response createAnnotation(Annotation annotation) {
+        AnnotationService annotationService = Framework.getService(AnnotationService.class);
+        Annotation result = annotationService.createAnnotation(getContext().getCoreSession(), annotation);
+        return Response.status(Response.Status.CREATED).entity(result).build();
+    }
+
+    @GET
+    public List<Annotation> getAnnotations(@QueryParam("xpath") String xpath) {
+        DocumentModel doc = getTarget().getAdapter(DocumentModel.class);
+        AnnotationService annotationService = Framework.getService(AnnotationService.class);
+        return annotationService.getAnnotations(getContext().getCoreSession(), doc.getId(), xpath);
+    }
+
+    @GET
+    @Path("{annotationId}")
+    public Annotation getAnnotation(@PathParam("annotationId") String annotationId) {
+        AnnotationService annotationService = Framework.getService(AnnotationService.class);
+        // The annotation id is enough to retrieve the data without its associated document id and xpath
+        return annotationService.getAnnotation(getContext().getCoreSession(), null, null, annotationId);
```

Comment:
Why not updating `AnnotationService#getAnnotation` method to not have `docId` and `xpath` as arguments as you don't need them? (you already updated `#createAnnotation` and `#updateAnnoation` methods).
Or at least add a new method without those arguments if you really want to keep the first one?
This will make the call easier to read and avoid the comment :)

Reply:
The idea was that we didn't want to enforce that the annotation id was globally unique. But in effect it's the case in the current implementation, so maybe YAGNI. Ok for this fix.